src/data/fintype/basic.lean

```diff
@@ -13,11 +11,58 @@ import data.array.lemmas
 import order.well_founded
 import group_theory.perm.basic
 
+/-!
+# Finite types
+
+This file defines a typeclass to state that a type is finite.
+
+## Main declarations
+
+* `fintype α`:  Typeclass saying that a type is finite. It takes as fields a `finset` and a proof
+  that all terms of type `α` are in it.
+* `finset.univ`: The finset of all elements of a fintype.
+* `fintype.card α`: Cardinality of a fintype. Also `univ.card`.
+* `perms_of_finset s`: The finset of permutations of the finset `s`.
+* `fintype.trunc_equiv_of_card_eq` `fintype.equiv_of_card_eq`
+* `fintype.trunc_equiv_fin`: A fintype `α` is computably equivalent to `fin (card α)`. The
+  `trunc`-free, noncomputable version is `fintype.equiv_fin`.
+* `fin.equiv_iff_eq`: `fin m ≃ fin n` iff `m = n`.
+* `infinite α`: Typeclass saying that a type is infinite. Defined as `fintype α → false`.
+* `not_fintype`: No `fintype` is infinite.
+* `infinite.nat_embedding`: An embedding of `ℕ` into an infinite type.
+
+We also provide the following versions of the pigeonholes principle.
+* `fintype.exists_ne_map_eq_of_card_lt` and `is_empty_of_card_lt`: Finitely many pigeons and
+  pigeonholes. Weak formulation.
+* `fintype.exists_ne_map_eq_of_infinite`: Infinitely many pigeons in finitely many pigeonholes.
+  Weak formulation.
+* `fintype.exists_infinite_fiber`: Infinitely many pigeons in finitely many pigeonholes. Strong
+  formulation.
+Some more is to be found in `data.fintype.card_embedding`.
```

Comment:
```suggestion
Some more pigeonhole-like statements can be found in `data.fintype.card_embedding`.
```
just to make it clear this isn't a general statement about fintypes.

Reply:
https://observablehq.com/@bryangingechen/github-lean-doc-preview says the blank line is needed to format correctly.